src/security.js

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import jwt from 'jsonwebtoken';
+import { minimatch } from 'minimatch';
+import fetch from 'node-fetch';
+import { ContextHelper } from './context.js';
+import { RestError } from './rest-error.js';
+import { SecretsManager } from './secret.js';
+
+const BEARER_OFFSET = 7;
+
+/**
+ * @typedef {Object} AuthenticationRequirement
+ * @property {Array<string>} [allowedRoles]
+ * @property {Array<string>} [allowedPermissions]
+ */
+
+/**
+ * @typedef {Object} TokenRequest
+ * @property {string} spaceId the space for which the token will be generated
+ * @property {Array<string>} roleKeys the role keys for which the token should be generated
+ * @property {string} generator provides attribution for the key in the description
+ * @property {number} [expiresInMinutes] the number of minutes before the token expires (or 14 days)
+ */
+
+/**
+ * @typedef {Object} TokenPayload
+ * @property {Array<string>} permissions the permissions granted to this token
+ * @property {Array<string>} roles the permissions granted to this token
+ * @property {string} sub the subject (or identifier) for this token
+ * @property {string} tenantId the primary tenant for the token
+ * @property {Array<string>} [tenantIds] the optional list of additional
+ *  allowed tenants for the token
+ * @property {string} type the type of the token
+ * @property {number} exp the timestamp at which the token will expire
+ */
+
+export class Security {
+  /**
+   * Checks to see if any of the allowed permissions are present in the actual permissions,
+   * using globbing expansion
```

Comment:
so actualPermissions must be a subset of allowedPermissions?

maybe add an ex for clarity